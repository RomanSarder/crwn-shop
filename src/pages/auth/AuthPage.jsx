import React from 'react'
import { useSelector } from 'react-redux'
import { Redirect } from 'react-router-dom'
import styled from 'styled-components'
import SignIn from '../../components/sign-in/SignIn'
import SignUp from '../../components/sign-up/SignUp'

import { selectUser } from '../../store/user/selectors'

const StyledAuthPage = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;

    @media (min-width: 1200px) {
        flex-direction: row;
        justify-content: space-around;
        align-items: flex-start;
    }
`

export default function AuthPage() {
    var currentUser = useSelector(selectUser)

    if (currentUser) {
        return <Redirect to="/" />
    } else {
        return (
            <StyledAuthPage>
                <SignIn />
                <SignUp />
            </StyledAuthPage>
        )
    }
}
