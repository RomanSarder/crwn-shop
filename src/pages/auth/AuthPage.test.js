import { render, screen } from '@testing-library/react'
import 'jest-styled-components'
import { Provider } from 'react-redux'
import { Router, Route, Switch } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import configureStore from 'redux-mock-store'

import AuthPage from './AuthPage'
import e from 'express'

// jest.mock('../../components/sign-in/SignIn', () => () => (<div>Sign In</div>))
// jest.mock('../../components/sign-up/SignUp', () => () => (<div>Sign Up</div>))

var history = createMemoryHistory()

function renderAuthPage (store) {
    var renderResult = render(
        <Router history={history}>
            <Switch>
                <Route path="/auth" exact component={() => (
                    <Provider store={store}>
                        <AuthPage/>
                    </Provider>
                )}/>
            </Switch>
        </Router>
    )

    return renderResult
}

it('<AuthPage /> should stay if no user profile is set', () => {
    history.push('/auth')
    var store = configureStore()({
        user: {
            currentUser: null
        }
    })
    var { getAllByRole, container } = renderAuthPage(store)

    var signbuttons = getAllByRole('button')
    expect(container).toMatchSnapshot()
    expect(signbuttons.length).toEqual(3)
    expect(signbuttons[0]).toHaveTextContent('Sign In')
    expect(signbuttons[1]).toHaveTextContent('Sign In With Google')
    expect(signbuttons[2]).toHaveTextContent('Sign Up')
    expect(history.location.pathname).toEqual('/auth')
})

it('<AuthPage /> redirect if user profile is set', () => {
    history.push('/auth')
    var store = configureStore()({
        user: {
            currentUser: { test: true }
        }
    })
    renderAuthPage(store)
    expect(history.location.pathname).toEqual('/')
})