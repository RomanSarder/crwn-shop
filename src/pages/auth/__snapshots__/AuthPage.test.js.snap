// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthPage /> should stay if no user profile is set 1`] = `
.c2 {
  position: relative;
  margin: 4.5rem 0;
}

.c4 {
  color: grey;
  font-size: 1.6rem;
  font-weight: normal;
  position: absolute;
  pointer-events: none;
  left: 0.5rem;
  top: 1rem;
  -webkit-transition: 300ms ease all;
  transition: 300ms ease all;
}

.c5 {
  background: none;
  background-color: white;
  color: grey;
  font-size: 1.8rem;
  padding: 1rem 1rem 1rem 0.5rem;
  display: block;
  width: 100%;
  border: none;
  border-radius: 0;
  border-bottom: 1px solid grey;
  margin: 2.5rem 0;
}

.c5:focus {
  outline: none;
}

.c5:focus ~ .c3 {
  top: -1.4rem;
  font-size: 1.2rem;
  color: black;
}

.c6 {
  min-width: 16.5rem;
  width: auto;
  height: 5rem;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 5rem;
  padding: 0 3.5rem 0 3.5rem;
  font-size: 1.5rem;
  background-color: black;
  color: white;
  text-transform: uppercase;
  font-family: 'Open Sans Condensed';
  font-weight: bolder;
  cursor: pointer;
  border: none;
}

.c6:hover {
  background-color: white;
  color: black;
  border: 1px solid black;
}

.c7 {
  min-width: 16.5rem;
  width: auto;
  height: 5rem;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 5rem;
  padding: 0 3.5rem 0 3.5rem;
  font-size: 1.5rem;
  background-color: black;
  color: white;
  text-transform: uppercase;
  font-family: 'Open Sans Condensed';
  font-weight: bolder;
  cursor: pointer;
  border: none;
  background-color: #4285f4;
  color: white;
}

.c7:hover {
  background-color: white;
  color: black;
  border: 1px solid black;
}

.c7:hover {
  background: #357ae8;
  border: none;
  color: white;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 .title {
  font-size: 2.8rem;
  margin: 1rem 0;
}

.c1 span {
  font-size: 2rem;
}

.c1 .buttons {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:576px) {
  .c1 {
    width: 38rem;
  }
}

@media (min-width:1200px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="title"
      >
        I already have an account
      </h2>
      <span>
        Sign In
      </span>
      <form>
        <div
          class="c2"
        >
          <label
            class="c3 c4"
            for="email-1"
          >
            Email
          </label>
          <input
            class="c5"
            id="email-1"
            name="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="c2"
        >
          <label
            class="c3 c4"
            for="password-2"
          >
            Password
          </label>
          <input
            class="c5"
            id="password-2"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="buttons"
        >
          <button
            class="c6"
            type="submit"
          >
            Sign In
          </button>
          <button
            class="c7"
            type="button"
          >
            Sign In With Google
          </button>
        </div>
      </form>
    </div>
    <div
      class="c1"
    >
      <h2
        class="title"
      >
        Register new account
      </h2>
      <span>
        Sign Up
      </span>
      <form>
        <div
          class="c2"
        >
          <label
            class="c3 c4"
            for="displayName-3"
          >
            Name
          </label>
          <input
            class="c5"
            id="displayName-3"
            name="displayName"
            type="text"
            value=""
          />
        </div>
        <div
          class="c2"
        >
          <label
            class="c3 c4"
            for="email-4"
          >
            Email
          </label>
          <input
            class="c5"
            id="email-4"
            name="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="c2"
        >
          <label
            class="c3 c4"
            for="password-5"
          >
            Password
          </label>
          <input
            class="c5"
            id="password-5"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="c2"
        >
          <label
            class="c3 c4"
            for="confirm-6"
          >
            Confirm Password
          </label>
          <input
            class="c5"
            id="confirm-6"
            name="confirm"
            type="password"
            value=""
          />
        </div>
        <button
          class="c6"
          type="submit"
        >
          Sign Up
        </button>
      </form>
    </div>
  </div>
</div>
`;
